### PR TITLE
run block validation for tipsets in parallel

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: chain/sync.go [hunk 1]
```diff
@@ -466,16 +466,25 @@ func (syncer *Syncer) ValidateTipSet(ctx context.Context, fts *store.FullTipSet)
 		return nil
 	}
 
+	var futures []async.ErrorFuture
 	for _, b := range fts.Blocks {
```
Comment (Kubuxu) [inline chain/sync.go:470]: First time in a long time, this lint warning saves the day.

`b` here is a reference, that will get updated to new value every loop iteration, and it is not captured correctly by closure.
Closure captures the reference `b`, not a copy `b`.
This effect could lead to us not validating all blocks.

The solution is to rebind `b` to a scoped variable:
```
b := b
```

Comment (Kubuxu) [reply chain/sync.go:470]: Or pass it as an argument to the closure but that is not simple here.

Comment (Kubuxu) [reply chain/sync.go:470]: Example: https://play.golang.org/p/yfHLJU5zkFZ

Comment (schomatis) [reply chain/sync.go:470]: :+1: for the clear explanation, go's syntactic sugar is evil

**Comment by schomatis** [comment]
@magik6k @Kubuxu In the future please document very opaque and deceivingly trivial but critical lines like
```
b := b
```
since it's not guaranteed that in the future whoever needs to operate on this part of the code will arrive at the explanation on this PR.

**Pull request merged**